dashboard/app/models/lesson.rb

```diff
@@ -291,6 +291,24 @@ def summarize(include_bonus_levels = false, for_edit: false)
     lesson_summary.freeze
   end
 
+  def summarize_for_calendar
+    activities = lesson_activities.map(&:summarize)
```

Comment:
I think this set of lines calculating duration can possibly be shrunk to one line with sum(https://apidock.com/rails/Enumerable/sum). If you can shrink it to one line then you can probably just put it inline in the returned object below